hackathon_site/dashboard/frontend/src/components/teamDetail/TeamPendingOrderTable/TeamPendingOrderTable.test.tsx

```diff
@@ -0,0 +1,71 @@
+import { fireEvent } from "@testing-library/react";
+
+describe("team pending order table", () => {
+    test("renders team pending order table", () => {
+        const { getByTestId } = render(<TeamPendingOrderTable />);
+        expect(screen.getByText("Requested Items")).toBeInTheDocument();
+
+        //loop through all pending orders
+        for (let i = 0; i < mockPendingOrdersInTable.length; i++) {
+            const currentOrder = mockPendingOrdersInTable[i];
+            //loop through all the hardware in each order
+            for (let j = 0; j < currentOrder.hardwareInTableRow.length; j++) {
+                // renders all hardware names
+                const currentRow = currentOrder.hardwareInTableRow[j];
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].name}`)
+                ).toBeInTheDocument();
+
+                // renders all hardware model numbers
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].model_number}`)
+                ).toBeInTheDocument();
+
+                // renders all hardware manufacturers
+                expect(
+                    within(
+                        getByTestId(`table-${currentOrder.id}-${currentRow.id}`)
+                    ).getByText(`${mockHardware[currentRow.id - 1].manufacturer}`)
+                ).toBeInTheDocument();
+            }
+            const submitButton = getByTestId(`complete-button-${currentOrder.id}`);
+            fireEvent.click(submitButton);
+        }
+    });
+
+    test("Complete Order button", async () => {
+        const { getByTestId } = render(<TeamPendingOrderTable />);
+
+        for (let i = 0; i < mockPendingOrdersInTable.length; i++) {
+            const currentOrder = mockPendingOrdersInTable[i];
+            const submitButton = getByTestId(`complete-button-${currentOrder.id}`);
+            fireEvent.click(submitButton);
+            await waitFor(() => {
+                for (let j = 0; j < currentOrder.hardwareInTableRow.length; j++) {
+                    const currentRow = currentOrder.hardwareInTableRow[j];
+                    expect(
+                        screen.getByText(
+                            `"${mockHardware[currentRow.id - 1].id}-quantity": "${
+                                currentRow.quantityGranted
+                            }"`
+                        )
+                    ).toBeInTheDocument();
+                    expect(
+                        screen.getByText(
+                            `"${mockHardware[currentRow.id - 1].id}-checkbox": false`
+                        )
+                    ).toBeInTheDocument();
+                }
+            });
+        }
```

Comment:
a test you could add tho is pressing the "All" button will change the dropdown value

Reply:
you can also test whether the checkbox actually checks the box (but that's a bit more complicated, I'll leave that as optional)

Reply:
> a test you could add tho is pressing the "All" button will change the dropdown value

should i set the default value to 0 then, so that I can test how it changes?

Reply:
Try to go about it as how the user would:

- when table is rendered dropdown is at max value
- User changes dropdown to lower value
- user clicks "All" and it should update the dropdown value